### DISCUSSION TITLE
Enabling dropout during trainer.predict

### DISCUSSION THREAD

**Discussion by standard-aaron** [Lightning-AI/pytorch-lightning#11710] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I want to enable dropout during `.predict` and tried implementing the following:

```
model.eval() 
for m in model.modules():
    if m.__class__.__name__.startswith('Dropout'):
        m.train()
                
...

trainer.predict(
    model, dataloaders=data_loader, return_predictions=True
)
```

It seems like `.predict` is overriding this because I get identical predictions with different seeds. 

Can someone explain how to accomplish this, or point me to the relevant docs? (Couldn't find them & tried looking for while) 

Thank you!

**Comment by rohitgr7** [comment, marked as answer]
hey @35ajstern !

you can enable this inside `predict_step` itself. Check this out: https://github.com/PyTorchLightning/pytorch-lightning/blob/f35e2210e240b443fd4dafed8fe2e30ee7d579ea/docs/source/common/production_inference.rst#prediction-api

this is part of a PR, will be available in the docs once merged.

**Reply by standard-aaron** [reply]
Thank you @rohitgr7!
